util/network/netproviders/network_windows.go

```diff
@@ -0,0 +1,53 @@
+// +build windows
+
+package netproviders
+
+import (
+	"os"
+
+	"github.com/moby/buildkit/solver/pb"
+	"github.com/moby/buildkit/util/network"
+	"github.com/moby/buildkit/util/network/cniprovider"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Opt struct {
+	CNI  cniprovider.Opt
+	Mode string
+}
+
+// Providers returns the network provider set
+func Providers(opt Opt) (map[pb.NetMode]network.Provider, error) {
```

Comment:
This looks quite similar to unix version. Wonder if we actually need the duplicate code and can just do a os check for host mode.

Reply:
That was my first cut, but it looks messy. I can probably refactor this easily enough.

Reply:
It just occurred to me that since we're only assembling things to pass to containerd, not actually manipulating data ourselves, do we need to worry about supporting a real Linux-based containerd, e.g. in WSL2 for the Docker Desktop use-case, from a Windows build of BuildKit? I'm assuming not, and that BuildKit is built to assume buildkitd is running on the same host as containerd and the containers themselves.

Reply:
You can ignore wsl2 case atm and assume buildkit is also running on Linux side in that case.

Reply:
If you still think os checks will be messy in the end then that is fine. Maybe paste here that version so it's limitations are visible.

Reply:
Well, thinking about it, can we treat 'host mode' as an invalid mode on Windows builds? That would simplify things, see the TODO later in the file:
```
	return map[pb.NetMode]network.Provider{
		pb.NetMode_UNSET: defaultProvider,
		// TODO: This is clearly a lie. I think the API requires this be filled in though.
		pb.NetMode_HOST: defaultProvider,
		pb.NetMode_NONE: network.NewNoneProvider(),
	}, nil
```

I reckon I can make it neater by reinstating the single `func Providers` and using helper functions for a couple of the cases instead. It was more that my _first cut_ wasn't neat.

This commit isn't essential to the flow, it's more of a sharp-edges fix because the host-mode networking setup will (unintentionally) put you in LCOW mode, even though the value it sets is not used in LCOW mode.

But getting this right will guide the fixes for `GenerateSpec`, which is going to be a much bigger set of "Very similar between Windows and Linux, except with a bunch of `runtime.GOOS` checks".

Reply:
If host is not supported it is fine to leave it out. I just thought a change like

```
host := network.NewHostProvider()
if runtime.GOOS == `windows` {
  host = nil
}
```

is simpler in this case.

Could also update executor to error out there is `host` is requested. Either by `GOOS` or by detecting that valid value does not exist in the map that is returned from here.

Reply:
It came out okay on my second pass. The executors already correctly reject missing map elements, so I didn't need to change anything there to make host-mode invalid on Windows.